hack/verify-structured-logging.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script is used to avoid regressions after a package is migrated
+# to structured logging. once a package is completely migrated add
+# it .structured_logging file to avoid any future regressions.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+source "${KUBE_ROOT}/hack/lib/util.sh"
+
+migrated_packages_file="${KUBE_ROOT}/hack/.structured_logging"
+kube::util::check-file-in-alphabetical-order "${migrated_packages_file}"
+
+migrated_packages=()
+while IFS='' read -r line; do
+  migrated_packages+=("$line")
+done < <(cat "${migrated_packages_file}")
+
+# Install logcheck
+pushd "${KUBE_ROOT}/hack/tools" >/dev/null
+  GO111MODULE=on go install k8s.io/klog/hack/tools/logcheck
```

Comment:
this may also need https://github.com/kubernetes/kubernetes/blob/792b4b1e32394a13c27741f71f14dd134d663a5f/hack/verify-govet-levee.sh#L30-L31

we don't want to install to the global GOBIN

Reply:
also needs a call to `kube::golang::setup_env` earlier I think (which will also validate the required go version if running this script by itself).

Reply:
Yes, the setup_env was required for me to run this locally.

Reply:
Using `kube::golang::setup_env` not working for me locally,  added `kube::golang::verify_go_version`, also I have checked other linting scripts they are not calling `setup_env`.
https://github.com/kubernetes/kubernetes/blob/master/hack/verify-staticcheck.sh

Reply:
`kube::golang::setup_env` is pretty standard, this is a bug in verify-staticcheck if anything https://cs.k8s.io/?q=kube%3A%3Agolang%3A%3Asetup_env&i=nope&files=&excludeFiles=&repos=